core/database_api.php

```diff
@@ -409,36 +409,46 @@ function db_query( $p_query, array $p_arr_parms = null, $p_limit = -1, $p_offset
 		$t_lastoffset = 0;
 		$i = 0;
 		if( !empty( $p_arr_parms ) ) {
-			while( preg_match( '/\?/', $p_query, $t_matches, PREG_OFFSET_CAPTURE, $t_lastoffset ) ) {
-				$t_matches = $t_matches[0];
+			# For mysql, tokens are '?', and parameters are binded sequentially
+			# For pgsql, tokens are '$number', and parameters are binded by the denoted index (1-based) in the parameter array
+			# For oracle, tokens are ':string', but ADOdb treat them as sequential in appearance order, so they behave like mysql
+			while( preg_match( '/(?<token>\?|\$|:)(?<index>[0-9]*)/', $p_query, $t_matches, PREG_OFFSET_CAPTURE, $t_lastoffset ) ) {
```

Comment:
Wrap long lines to avoid horizontal scrolling.
For the while loop, maybe set `$t_regex '/(?<token>\?|\$|:)(?<index>[0-9]*)/';` 
then `while( preg_match( $t_regex, ... `